tests/conftest.py

```diff
@@ -149,3 +163,112 @@ def DMGE(helpers: Helpers) -> DataModelGraphExplorer:
     """Fixture to instantiate a DataModelGraphExplorer object."""
     dmge = helpers.get_data_model_graph_explorer(path="example.model.jsonld")
     return dmge
+
+
+@pytest.fixture(scope="function")
+def schedule_for_cleanup(
+    request, synapse_store: SynapseStorage
+) -> Callable[[CleanupItem], None]:
+    """Returns a closure that takes an item that should be scheduled for cleanup."""
+
+    items: Set[CleanupItem] = set()
+
+    def _append_cleanup(item: CleanupItem):
+        print(f"Added {item} to cleanup list")
+        items.add(item)
+
+    def cleanup_scheduled_items() -> None:
```

Comment:
Nice custom function here, I know synapseclient has a cleanup utility in its tests, not sure if it's usable here.

Reply:
I used that as the basis for this function here. I tried to simplify this down a bit to handle for a narrower set of tasks.